### DISCUSSION TITLE
Composition typescript error accessing `this` on `computed`

### DISCUSSION THREAD

**Discussion by iyume** [vuejs/pinia#2623] [Help and Questions] [answered]
Related: https://github.com/vuejs/pinia/discussions/1151

I'm using composition api to define store. Is there any way to access `this`?

![图片](https://github.com/vuejs/pinia/assets/48091591/dc3240e3-cd32-4ade-908e-581fdc873502)

![图片](https://github.com/vuejs/pinia/assets/48091591/20e91210-915d-4b27-af55-db2d3ca8d4c2)

**Comment by ilyarefraction** [comment, marked as answer]
Why do you even need `this` in setup stores? If the `getNodeData` method is defined somewhere above, just access it as any local identifier:
```ts
    selectedNodeData: computed(() => getNodeData)
```
If you mean you need **component**'s instance which utilizes your composable you can use [getCurrentInstance](https://v3.ru.vuejs.org/api/composition-api.html#getcurrentinstance) api, but this is definitely a bad practice and I don't recommend doing so.

**Reply by posva** [reply]
As a note, `getCurrentInstance()` is not guaranteed to exist. It depends on how the store is first instantiated

**Comment by iyume** [comment]
Thanks! The `getNodeData` is not defined above but in the return braces. So I need to define these code in main function body and re-declaration them in return braces. Anyway thanks to your reply!
